src/net_processing.cpp

```diff
@@ -301,6 +370,9 @@ struct CNodeState {
     }
 };
 
+// Keeps track of the time (in microseconds) when transactions were requested last time
+limitedmap<uint256, int64_t> m_already_asked_for(MAX_INV_SZ);
```

Comment:
> For some reason I didn't manage to make it work with limitedmaps. It seemed like it's either size allocation or GUARDED_BY, and I don't see how to allocate it anywhere else after my code changes.


This might work

```suggestion
limitedmap<uint256, int64_t> m_already_asked_for GUARDED_BY(cs_main)(MAX_INV_SZ);
```